lib/extractor/docker-archive/index.ts

```diff
@@ -34,3 +38,32 @@ export function getPlatformFromConfig(
     ? `${imageConfig.os}/${imageConfig.architecture}`
     : undefined;
 }
+
+export function getDetectedLayersInfoFromConfig(imageConfig) {
+  const runInstructions = getUserInstructionLayersFromConfig(imageConfig)
+    .filter((instruction) => !instruction.empty_layer)
+    .map((instruction) => instruction.created_by);
+
+  const packages = getPackagesFromRunInstructions(runInstructions);
+  const layers = getLayersFromPackages(packages);
+  return { packages, layers };
+}
+
+export function getUserInstructionLayersFromConfig(imageConfig) {
+  const diffInHours = (d1, d2) => Math.abs(d1 - d2) / 1000 / (60 * 60);
+  const maxDiffInHours = 5;
+
+  const history = imageConfig.history;
+  const lastInstructionTime = new Date(history.slice(-1)[0].created);
```

Comment:
does that `-1` work in node? #til!

Reply:
one last question on that :) any chance that `imageConfig.history` can be empty? should we check that?

Reply:
That's a good question. I'm not sure. I want to say that it doesn't make sense that it would be empty, but worth a check. Will add!